library/src/main/java/net/vrgsoft/library/LinkCrawler.kt

```diff
@@ -79,42 +83,49 @@ class LinkCrawler {
                         content.description = ""
                     }
                     else -> try {
-                        val doc: Document = Jsoup.connect(content.finalUrl).userAgent("Mozzila").get()
+                        val doc = Jsoup.connect(content.finalUrl)
+                            .userAgent("Mozzila")
+                            .get()
                         content.htmlCode = extendedTrim(doc.toString())
+
                         val metaTags: Map<String, String> = getMetaTags(content.htmlCode)
                         content.metaTags = metaTags
-                        content.title = metaTags["title"]!!
-                        content.description = metaTags["description"]!!
-
-                        when {
-                            content.title == "" -> {
-                                val matchTitle = Regex.match(content.htmlCode, Regex.TITLE_PATTERN, 2)
-                                if (matchTitle != "") {
-                                    content.title = htmlDecode(matchTitle)
-                                }
-                            }
+                        content.title = metaTags["title"] ?: ""
+                        content.description = metaTags["description"] ?: ""
+
+                        if(content.title.isEmpty()) {
+                              val matchTitle = Regex.match(
+                                  content = content.htmlCode,
+                                  pattern = TITLE_PATTERN,
+                                  index = 2
+                              )
+                              if (matchTitle.isNotEmpty()) {
+                                  content.title = htmlDecode(matchTitle)
+                              }
                         }
-                        if (content.description == "") {
+                        if (content.description.isEmpty()) {
                             content.description = crawlCode(content.htmlCode)
                         }
                         content.description = content.description.replace(Regex.SCRIPT_PATTERN, "")
-                        when {
-                            metaTags["image"] != "" -> content.images.add(
-                                    metaTags["image"]!!)
-                            else -> content.images = getImages(doc).toMutableList()
+
+                        if(metaTags["image"]?.isNotEmpty() == true) {
```

Comment:
```suggestion
                        if (metaTags["image"]?.isNotEmpty() == true) {
```

Reply:
@MSF-Jarvis that's not valid syntax, the `?` makes it return `Boolean?` so an equality check is necessary

Reply:
> @MSF-Jarvis that's not valid syntax, the `?` makes it return `Boolean?` so an equality check is necessary

Oh true, didn't notice the nullable type. Will cut this down to just the formatting.